modules/post/multi/gather/memory_search.rb

```diff
@@ -0,0 +1,226 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Memory Search',
+        'Description' => %q{
+          This module allows for searching the memory space of running processes for
+          potentially sensitive data such as passwords.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => %w[sjanusz-r7],
+        'SessionTypes' => %w[meterpreter],
+        'Platform' => %w[linux unix osx windows],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_sys_process_memory_search
+              stdapi_sys_process_get_processes
+            ]
+          }
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        ::Msf::OptString.new('PROCESS_NAMES_GLOB', [false, 'Glob used to target processes', 'ssh.*']),
+        ::Msf::OptString.new('PROCESS_IDS', [false, 'Comma delimited process ID/IDs to search through']),
```

Comment:
It would be worth validating this with a regex. Something like `regex:/^(\s*\d(\s*,\s*\d+\s*)*)?$/` should do the trick while ignoring whitespace.